src/main/java/com/google/sps/data/UserStorage.java

```diff
@@ -27,50 +27,72 @@
 import java.util.ArrayList;
 
 public class UserStorage {
-  public static User getUser(String user_id) {
-    // TODO: Query in datastore.
-    return null;
-  }
-
-  public static User getUserByUsername(String username) {
-    // TODO: Query in datastore.
+  public static User getUser(String userId) {
+    Query query = new Query("User").setFilter(new FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL, KeyFactory.createKey("User", userId)));
+    Entity userEntity = DatastoreServiceFactory.getDatastoreService().prepare(query).asSingleEntity();
+    if (userEntity != null) {
+      return new User(userId, 
+        (String) userEntity.getProperty("email"),
+        (String) userEntity.getProperty("username"),
+        (ArrayList) userEntity.getProperty("invited-events"),
+        (ArrayList) userEntity.getProperty("joined-events"),
+        (ArrayList) userEntity.getProperty("declined-events")
+      );
+    }
     return null;
   }
 
   public static void addUser(User user) {
+    if (user == null)
+      return;
+
     // Make an Entity of user.
     Entity userEntity = new Entity("User", user.getID());
 
+    userEntity.setProperty("email", user.getEmail());
+    userEntity.setProperty("username", user.getUsername());
+    userEntity.setProperty("invited-events", user.getInvitedEventsID());
+    userEntity.setProperty("joined-events", user.getJoinedEventsID());
+    userEntity.setProperty("declined-events", user.getDeclinedEventsID());
+    
     // Store Entities to datastore.
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(userEntity);
+    DatastoreServiceFactory.getDatastoreService().put(userEntity);
   }
 
   public static void editUser(User user) {
```

Comment:
Not clear to me how the editing actually happens

Reply:
Now previous entity of user with this id is deleted and then new copy adding.